src/main/java/randoop/reflection/OmitMethodsPredicate.java

```diff
@@ -0,0 +1,229 @@
+package randoop.reflection;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+import java.util.regex.Pattern;
+import randoop.operation.TypedClassOperation;
+import randoop.types.ClassOrInterfaceType;
+import randoop.types.JavaTypes;
+import randoop.util.Log;
+
+/**
+ * Tests whether an operation matches a user-specified pattern, indicating that the operation should
+ * be omitted from the operation set.
+ *
+ * <p>An operation matches a pattern if the pattern matches the {@link RawSignature} of the
+ * operation or the inherited operation in a superclass. This class provides methods that (1) test
+ * the raw signature of an operation, and (2) test the raw signature of an operation and that of the
+ * same operation in superclasses.
+ */
+public class OmitMethodsPredicate {
+
+  /** An OmitMethodsPredicate that does no omission. */
+  public static OmitMethodsPredicate NO_OMISSION = new OmitMethodsPredicate(null);
+
+  /** {@code Pattern}s that match operations that should be omitted. */
+  private final List<Pattern> omitPatterns;
+
+  /**
+   * If {@code omitPatterns} is null, treat it as the empty list.
+   *
+   * @param omitPatterns a list of regular expressions for method signatures, or null
+   */
+  public OmitMethodsPredicate(List<Pattern> omitPatterns) {
+    if (omitPatterns == null) {
+      this.omitPatterns = new ArrayList<>();
+    } else {
+      this.omitPatterns = new ArrayList<>(omitPatterns);
+    }
+  }
+
+  /**
+   * Returns true if the operation is a constructor or method call and some omit pattern matches the
+   * {@link RawSignature} of the operation.
+   *
+   * @param operation the operation to match against the omit patterns of this predicate
+   * @return true if the signature matches an omit pattern, and false otherwise
+   */
+  boolean shouldOmit(TypedClassOperation operation) {
+    // Done if there are no patterns
+    if (omitPatterns.isEmpty()) {
+      return false;
+    }
+    // Only match constructors or methods
+    if (!operation.isConstructorCall() && !operation.isMethodCall()) {
+      return false;
+    }
+
+    String signature = operation.getRawSignature().toString();
+
+    for (Pattern pattern : omitPatterns) {
+      boolean result = pattern.matcher(signature).find();
+      if (Log.isLoggingOn()) {
+        Log.logLine(
+            String.format(
+                "Comparing '%s' against pattern '%s' = %b%n", signature, pattern, result));
+      }
+      if (result) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Indicates whether an omit pattern matches the raw signature of the method in either the
+   * declaring class of the method or a supertype.
```

Comment:
What is the relationship between "declaring class of the method" and "the type to which the method belongs"?  Please clarify, and use consistent terminology when possible.